R/compare_results.R

```diff
@@ -44,7 +44,12 @@ sig_compare <- function(sig1, sig2, metric = c("cosine", "jsd"),
   comparison$y_sig_index <- as.numeric(comparison$y_sig_index)
   comparison <- comparison[order(comparison[[metric_name]], decreasing = TRUE), 
                            ]
-  return(comparison)
+  split_comparison <- split(comparison, comparison$x_sig_name)
```

Comment:
We still want to be able to do the old way of plotting as well. Could you add a parameter to this function like this:

`type = c("best", "threshold")`

If "best" is selected, then you can run your new code. If "threshold" is selected, you can run the old way using the threshold parameter. Note that you will also have to add this parameter to the other compare-* functions

Reply:
Also, make sure to document the new parameter!